src/plots/cartesian/constraints.js

```diff
@@ -57,10 +57,7 @@ exports.handleConstraintDefaults = function(containerIn, containerOut, coerce, o
     // 'matches' wins over 'scaleanchor' (for now)
     var scaleanchor, scaleOpts;
 
-    if(!matches &&
```

Comment:
@alexcjohnson so I loosened this check, but I also had to remove the check below ... is this a belt-and-suspenders situation? If I restore the check below such that we only remove scale if it's the same letter as matches, does that seem sufficient?